src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -256,18 +295,17 @@ def _get_components(self):
         # components is necessary
 
         start_time = time.perf_counter()
+        pt_cache = FASTGAHEPowerTrainConfigurator._cache[self._power_train_file]
 
-        if self._components_id is None:
+        if self._components_id is None and not pt_cache.get("get_component_time"):
```

Comment:
This is how the presence of the components in the cache is checked ?
Plus I suggest you change the if to:
```python
if not self._components_id and not pt_cache.get("get_component_time"):
```
The original implementation was not very pythonic